### PR TITLE
Update for PROJ 7.0.0

### PR DESCRIPTION
**Pull request by hobu**
Checklist
* [x] Used a [fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by mwtoews** [comment]
@hobu I have a commit that you can cherry-pick here 091021ae7d5d3c989b7b88303d042c2934183b0e

File: recipe/meta.yaml
Comment (mwtoews) [inline recipe/meta.yaml]: should `datumgrid_ver` be renamed to `data_ver`?

**Comment by xylar** [comment]
The old `proj-datumgrid` data was about 6 MB, whereas the new `proj-data` is about 480 MB.  I just want to make sure we want to package that full data set into `proj` as opposed to having an optional, separate package.

**Comment by xylar** [comment]
```yaml
test:
  commands:
    # Check if NAD27/NAD83 datum shifting is present if `conus` is present.
    - test -f $PREFIX/share/proj/conus  # [not win]
    - if not exist %LIBRARY_PREFIX%\\share\\proj\\conus exit 1  # [win]
```
These tests no longer work because the `conus` file isn't part of `proj-data`.  Do we want to replace this test or just remove it?

File: recipe/meta.yaml
Comment (xylar) [inline recipe/meta.yaml]: According to the [docs](https://www.osgeo.org/foundation-news/proj-7-0-0/), this should be `libcurl` instead of `curl`, right?

**Comment by xylar** [comment]
@hobu, I went ahead and pushed some changes to your branch.  I hope that's okay with you.

File: recipe/meta.yaml
Comment (xylar) [inline recipe/meta.yaml:41]: Do we want to replace these tests?  If so, what with?

File: recipe/meta.yaml
Comment (xylar) [reply recipe/meta.yaml]: I went ahead and made this change.

File: recipe/meta.yaml
Comment (ocefpaf) [reply recipe/meta.yaml:41]: We probably don't need that. These tests were from before we packaged the datum grids and they probably belong in those feedstock instead of here.

File: recipe/meta.yaml
Comment (ocefpaf) [inline recipe/meta.yaml:33]: :+1:

**Review by ocefpaf — APPROVED** [review]

**Comment by ocefpaf** [comment]
Thanks @hobu for the PR and @mwtoews and @xylar for the review!

@hobu do you want to add yourself as a maintainer here so you can merge this in the future?

**Comment by xylar** [comment]
@ocefpaf, what is your opinion on including `proj-data` (480 MB) vs. having it as a separate package a la `proj-datumgrid-*`?  It seems like the `proj` package should work without `proj-data`, and users should have the option to keep the env. small if they want to.

**Comment by snowman2** [comment]
With the [projsync](https://proj.org/apps/projsync.html) application, I think including the grids is not as important.

**Comment by xylar** [comment]
Thanks @snowman2.  I'm working on a new recipe for `proj-data` on staged recipes.  Once I have that in, I'll remove it from being packaged here.

**Comment by ocefpaf** [comment]
@snowman2 is way better informed than me to answer that. I had no idea about `projsync` and I'm :+1: on keep the package small.

**Comment by hobu** [comment]
Thanks for the updates @xylar.

I think the Conda Forge recipes should consider enabling `PROJ_NETWORK=ON`, but that is maybe a wider discussion. Of course, you actually have to have network access for that to work, but for most people, it is going to be the most convenient require the least amount of storage. Where this will be the most effective is in notebook and lambda-style scenarios, which is one of the reasons I have been organizing its development.

**Comment by hobu** [comment]
I guess I should provide a little more detail:

`PROJ_NETWORK=ON` enables on-demand caching of the grid files from a Cloudfront-based CDN at https://cdn.proj.org/

Details about this capability can be found at https://proj.org/community/rfc/rfc-4.html

**Comment by xylar** [comment]
@hobu, what happens if we set `PROJ_NETWORK=ON` (e.g when the environment gets enabled) but the user:
1. doesn't have write access to `${PREFIX}/share/proj`?
2. doesn't have network access?
Presumably there are errors, which might still be fine if we can somehow point users to the need to install the `proj-data` package?

**Comment by ocefpaf** [comment]
> I guess I should provide a little more detail:
> 
> `PROJ_NETWORK=ON` enables on-demand caching of the grid files from a Cloudfront-based CDN at [cdn.proj.org](https://cdn.proj.org/)
> 
> Details about this capability can be found at [proj.org/community/rfc/rfc-4.html](https://proj.org/community/rfc/rfc-4.html)

Nice. Did not know that. We should check if our grid packages are installing in the same directories to avoid duplicating data in case the user downloads the grids from the packages.

**Comment by xylar** [comment]
> We should check if our grid packages are installing in the same directories to avoid duplicating data in case the user downloads the grids from the packages.

The documentation suggests the downloading tool will try to put things in `$PREFIX/share/proj`:
https://proj.org/apps/projsync.html#cmdoption-projsync-system-directory

This should be the same location where the data will end up with the `proj-data` package I'm working on.

**Comment by ocefpaf** [comment]
> The documentation suggests the downloading tool will try to put things in $PREFIX/share/proj

I'd rather if we moved away from the environment directory and adhere to XDG like directories.

**Comment by xylar** [comment]
> I'd rather if we moved away from the environment directory and adhere to XDG like directories.

Can you point me to some documentation on this?

**Comment by xylar** [comment]
Presumably as long as the data goes into the directory pointed to by `$PROJ_LIB`, this can go wherever we choose.

**Comment by ocefpaf** [comment]
> Can you point me to some documentation on this?

I believe this is the current docs on this topic:

https://specifications.freedesktop.org/basedir-spec/basedir-spec-latest.html

> Presumably as long as the data goes into the directory pointed to by $PROJ_LIB, this can go wherever we choose.

Indeed but is `projsync` reading that env var to store the data? If so we are good to go on our side and just change `$PROJ_LIB`.

**Comment by xylar** [comment]
> Indeed but is projsync reading that env var to store the data? If so we are good to go on our side and just change $PROJ_LIB.

Good question.  I can test this. @hobu, do you know the answer?

**Comment by xylar** [comment]
> I'd rather if we moved away from the environment directory and adhere to XDG like directories.

@ocefpaf, thanks for the docs.  Could you help me understand how this would look in this case?  What would we set $PROJ_LIB to in the activate scripts?

**Comment by ocefpaf** [comment]
> @ocefpaf, thanks for the docs. Could you help me understand how this would look in this case? What would we set $PROJ_LIB to in the activate scripts?

Actually these docs are probably better. We should try to get `$XDG_DATA_HOME` from the system, if that is not available we should define it with the default `$HOME/.local/share`. For proj's case that should be `$HOME/.local/share/proj`.

BTW, check cartopy for more info, they follow XDG directories nicely on *nix and Windows (which I have no idea what is the best strategy).

PS: while I'd like to implement that, please don't hold off merging stuff b/c of this! If this PR is done let's go ahead and merge it!

**Comment by xylar** [comment]
> PS: while I'd like to implement that, please don't hold off merging stuff b/c of this! If this PR is done let's go ahead and merge it!

This seems like a good time to do that, actually.  It could be confusing if this happens in build 1 of the same version.

**Comment by xylar** [comment]
I've taken `proj-data` out of the package and added `PROJ_NETWORK=ON` to the activate packages.

@hobu, could you check that this is seems right, as far as you can tell?

**Comment by xylar** [comment]
@ocefpaf, I don't think we're free to put files anywhere outside of the environment.  So if we want users to be able to install a `proj-data` package, they're going to have to point to a directory within the environment at least for that case.

For cases where data gets downloaded, I would agree it makes some sense for that data to go into a directory outside of the environment.  But that's tricky because there's some data from the `proj` package itself that goes into the `$PROJ_LIB` directory.  This means that, in some sense, we need a directory inside the environment (e.g. for `proj.db`) and a different one potentially outside of the environment for data from `projsync`.  It's not clear to me yet how to accomplish this, so I do think it might be something to explore later on and to merge this for now.

**Comment by hobu** [comment]
I think the combo of `PROJ_NETWORK=ON` and the recommendation to use `projsync` is enough to cover the common cases of how people would typically interact with the grid files. A convenience grid-data package may still be useful, I don't know.

Anything to add to this discussion @rouault?

**Comment by xylar** [comment]
@conda-forge-admin, please rerender

**Comment by xylar** [comment]
^^ Oops, pinnings are still for `curl`, not `libcurl`

**Comment by ocefpaf** [comment]
> @ocefpaf, I don't think we're free to put files anywhere outside of the environment. So if we want users to be able to install a `proj-data` package, they're going to have to point to a directory within the environment at least for that case.

Thanks for looking into it. While we could do a post-script build or something it would be too hacky. Let's keep them in the environment then. Sorry for the bad suggestion.

> For cases where data gets downloaded, I would agree it makes some sense for that data to go into a directory outside of the environment. But that's tricky because there's some data from the `proj` package itself that goes into the `$PROJ_LIB` directory. This means that, in some sense, we need a directory inside the environment (e.g. for `proj.db`) and a different one potentially outside of the environment for data from `projsync`. It's not clear to me yet how to accomplish this, so I do think it might be something to explore later on and to merge this for now.

Yeah. That would be confusing.

**Comment by xylar** [comment]
I'd like to merge this after the testing finishes (successfully).  If anyone would like met to hold off, please let me know within the next hour or so.

**Comment by hobu** [comment]
Pins won't be updated for a while, correct?

**Comment by xylar** [comment]
> Pins won't be updated for a while, correct?

The `conda-forge-pinning` repo gets updated fairly regularly, but this package will be updated more-or-less automatically every time they do (there will be a pull request to rebuild with updated pins).

**Comment by xylar** [comment]
Or are your referring to to the pin of `proj` itself?  That's something we can decide separate from this PR.

**Comment by xylar** [comment]
Thanks @hobu, @mwtoews and @ocefpaf for contributing to this and @snowman2 for input on `proj-data`.

**Pull request merged**
